### PR TITLE
Relax qdarkstyle version bound

### PR DESCRIPTION
**Pull request by MrNaif2018**
This PR removes hard pin on qdarkstyle, resolving conflicts between electrums
Electrum has unpinned it too, because they have removed the psutil and other dependencies, leaving only necessary ones (qtpy to work with any qt library, pyqt4/pyqt5/pyside5/pyside6, etc and packaging to parse versions and so on, a standard frequently-used python library)
There aren't much style changes as of now
See spesmilo/electrum@1ad30e86b6d1926d0ac5bf10a69288430d3df6c6
There are small visual changes, but I would say it is even more adequate now (i.e. no jumping on hover or ugly borders around buttons):
изображение
изображение
изображение

I've also updated deterministic build folder kind of partially manually via pip-tools, because freeze-packages modified the file heavily (btw I suggest using pip-tools for managing frozen dependencies)

### CODE REVIEW

**Comment by EchterAgo** [comment]
I haven't looked into this yet, but if you touch this you might want to check the style patches in `electroncash_gui/qt/style_patcher.py`.

There are some style patches depending on the QDarkStyle version that maybe are not needed anymore and should be conditional based on the QDarkStyle version.

**Comment by MrNaif2018** [comment]
Yeah I saw them, are the old patches for < 2.6.8 good to be removed now? Maybe it was left for repackagers?

**Comment by EchterAgo** [comment]
No, we shouldn't remove the old patches, but we might want to limit the version range of the `qdarkstyle >= 2.6.8` if those are not needed anymore with newer versions.

**Comment by MrNaif2018** [comment]
@EchterAgo I've updated the PR. Electrum doesn't have the logic for disabling dark theme on mac os, maybe it works now with recent qdarkstyle? Could someone with a mac test it?
I've backported their up-to-date style patches
Though I think that patches for qdarkstyle < 2.6.8, which is old and probably not used anywhere shouldn't be kept. @cculianu What do you think?

File: electroncash_gui/qt/__init__.py [hunk 1]
```diff
@@ -336,13 +336,7 @@ def is_dark_theme_available(self):
         return True
 
     def set_dark_theme_if_needed(self):
-        if sys.platform in ('darwin',):
-            # On OSX, qdarkstyle is kind of broken. We instead rely on Mojave
-            # dark mode if (built in to the OS) for this facility, which the
-            # user can set outside of this application.
-            use_dark_theme = False
-        else:
-            use_dark_theme = self.config.get('qt_gui_color_theme', 'default') == 'dark'
```
Comment (cculianu) [inline electroncash_gui/qt/__init__.py:345]: Why did you delete this?

**Comment by cculianu** [comment]
@MrNaif2018 I think we need to disable it on macOS.  macOS has its own dark style that looks FAR FAR FAR better.  Also, Electrum doesn't even work right on macOS to support native dark mode, which is why they use the uglier QDarkStyle...

Please restore back the old macOS code..

**Comment by MrNaif2018** [comment]
Sorry, didn't have a macbook to test. I thought that if it was enabled it should have been tested in electrum (:
Reverted back

**Comment by cculianu** [comment]
News flash: Electrum Ui is kind of crappy/broken/not as good as Electron Cash's.

**Comment by cculianu** [comment]
Thanks for reverting the macos stuff. I have to try the deterministic build now to see if the new packages play nicely with each other.. or did you test that already?

**Comment by MrNaif2018** [comment]
Well not really because it kind of failed for me or was upgrading all packages at once, so to keep diff small I ran pip-tools than manually carefully removed things that didn't need upgrading and to follow same naming etc

**Comment by cculianu** [comment]
Ok well I'll build windows and linux now and see if anything broke.  If it looks ok and nothing broke, I will merge it.

**Comment by cculianu** [comment]
I built windows and tested it. Look ok to me. I'm still waiting on linux, but I'll just merge this now and on the odd chance we need to fix things up from Linux, I'll take it from there.  I'm relying on the fact that you run Linux and it looked ok for you.
